Assets/Scripts/UI/Terminal/TerminalUIManager.cs

```diff
@@ -53,36 +53,73 @@ private void SetupButtonListeners()
     }
 
     private void TabButtonClicked(Tab tabClicked)
+    {
+        SetActivePanel(tabClicked);
+        SetTabButtonInteractability(tabClicked);
+    }
+
+    private void SetActivePanel(Tab tabClicked)
     {
         ClearPanels();
+        GetPanelByTabClicked(tabClicked).SetActive(true);
+    }
+
+    private void ClearPanels()
+    {
+        missionsPanel.SetActive(false);
+        messagesPanel.SetActive(false);
+        analyticsPanel.SetActive(false);
+        crewPanel.SetActive(false);
+        upgradesPanel.SetActive(false);
+        newDayReportPanel.SetActive(false);
+    }
+
+    private GameObject GetPanelByTabClicked(Tab tabClicked)
+    {
         switch (tabClicked)
         {
             case Tab.Missions:
-                missionsPanel.SetActive(true);
-                break;
+                return missionsPanel;
             case Tab.Messages:
-                messagesPanel.SetActive(true);
-                break;
+                return messagesPanel;
             case Tab.Analytics:
-                analyticsPanel.SetActive(true);
-                break;
+                return analyticsPanel;
             case Tab.Crew:
-                crewPanel.SetActive(true);
-                break;
+                return crewPanel;
             case Tab.Upgrades:
-                upgradesPanel.SetActive(true);
-                break;
+                return upgradesPanel;
+            default:
+                return null;
         }
     }
 
-    private void ClearPanels()
+    private void SetTabButtonInteractability(Tab tabClicked)
     {
-        missionsPanel.SetActive(false);
-        messagesPanel.SetActive(false);
-        analyticsPanel.SetActive(false);
-        crewPanel.SetActive(false);
-        upgradesPanel.SetActive(false);
-        newDayReportPanel.SetActive(false);
+        missionsButton.interactable = true;
+        messagesButton.interactable = true;
+        analyticsButton.interactable = true;
+        crewButton.interactable = true;
+        upgradesButton.interactable = true;
+        GetTabButtonByTabClicked(tabClicked).interactable = false;
```

Comment:
we probably don't want this. This should be purely a visual change. By making it not interactable.... What if they want to just refresh the page? They would have to click on a different tab and come back

Reply:
i did not even consider that refreshing the page would be a thing

Reply:
if they are in the mail page for example. Maybe they want to click on the tab to go back to the high level view, instead of the back button